BibBuddy/app/src/main/java/de/bibbuddy/NotesFragment.java

```diff
@@ -25,103 +24,96 @@
-
-    sortCriteria = ((MainActivity) getActivity()).getSortCriteria();
 
     noteModel = new NoteModel(getContext());
-    notes = noteModel.getCompleteNoteList();
-    setHasOptionsMenu(true);
-    setupRecyclerView();
-    enableSwipeToDelete();
+    notesRecyclerView = view.findViewById(R.id.note_list_recycler_view);
+    noteList = noteModel.getNoteList();
+    sortCriteria = ((MainActivity) requireActivity()).getSortCriteria();
+    emptyListView = view.findViewById(R.id.empty_notes_list_view);
 
-    ((MainActivity) getActivity()).setVisibilityImportShareButton(View.INVISIBLE, View.INVISIBLE);
+    ((MainActivity) requireActivity())
+        .setVisibilityImportShareButton(View.INVISIBLE, View.INVISIBLE);
     setupSortBtn();
+    setHasOptionsMenu(true);
+    setupRecyclerView();
 
     return view;
   }
 
-  private void setupSortBtn() {
-    ImageButton sortBtn = getActivity().findViewById(R.id.sort_btn);
-    ((MainActivity) getActivity()).setVisibilitySortButton(true);
-
-    sortBtn.setOnClickListener(new View.OnClickListener() {
-      @Override
-      public void onClick(View v) {
-        handleSortNote();
-      }
-    });
-  }
-
   @Override
   public void onCreateOptionsMenu(@NonNull Menu menu, MenuInflater inflater) {
     inflater.inflate(R.menu.fragment_note_list_menu, menu);
     super.onCreateOptionsMenu(menu, inflater);
   }
 
+  @Override
+  public void onPrepareOptionsMenu(Menu menu) {
+    MenuItem deleteNote = menu.findItem(R.id.menu_note_list_delete);
+    deleteNote.setVisible(adapter.getSelectedNoteItems().size() > 0);
+  }
 
   @Override
   public boolean onOptionsItemSelected(MenuItem item) {
-
-    switch (item.getItemId()) {
-
-      case R.id.menu_help_note_list:
-        handleManualNotesList();
-        break;
-
-      default:
-        Toast.makeText(getContext(), "Fehler", Toast.LENGTH_SHORT).show();
+    long id = item.getItemId();
+    if (id == R.id.menu_note_list_delete) {
+      handleDeleteNote();
+    } else if (item.getItemId() == R.id.menu_note_list_help) {
+      handleHelpNotesFragment();
+    } else {
+      Toast.makeText(getContext(), R.string.error, Toast.LENGTH_SHORT).show();
     }
 
     return super.onOptionsItemSelected(item);
   }
 
-  private void handleSortNote() {
-    SortDialog sortDialog = new SortDialog(getContext(), sortCriteria,
-        new SortDialog.SortDialogListener() {
-          @Override
-          public void onSortedSelected(SortCriteria newSortCriteria) {
-            sortCriteria = newSortCriteria;
-            ((MainActivity) getActivity()).setSortCriteria(newSortCriteria);
-            sortNoteList();
-          }
-        });
+  private void handleDeleteNote() {
+    AlertDialog.Builder alertDeleteBookNote = new AlertDialog.Builder(requireActivity());
 
-    sortDialog.show();
+    alertDeleteBookNote.setCancelable(false);
+    alertDeleteBookNote.setTitle(R.string.delete_note);
+    alertDeleteBookNote.setMessage(R.string.delete_note_message);
```

Comment:
Please make a distinction here between deleting a single or multiple notes here like in BookNotesView.java and formulate the error message in the same way.

```suggestion
 if (adapter.getSelectedNoteItems().size() > 1) {
      alertDeleteBookNote.setTitle(R.string.delete_notes);
      alertDeleteBookNote.setMessage(
          getString(R.string.delete_notes_message) + " " + getString(R.string.delete_warning));
    } else {
      alertDeleteBookNote.setTitle(R.string.delete_note);
      alertDeleteBookNote.setMessage(
          getString(R.string.delete_note_message) + " " + getString(R.string.delete_warning));
    }
```

Reply:
Done.